### PR TITLE
[20.09] firejail: 0.9.62 -> 0.9.64.4

### LINKED ISSUE #112423: Security: please update firejail to 0.9.64.4 to fix root privilege escalation in OverlayFS code (CVE-2021-26910)
**Describe the bug**

 https://seclists.org/oss-sec/2021/q1/121

> Summary: A vulnerability resulting in root privilege escalation was discovered in Firejail's OverlayFS code,
> 
> Versions affected: Firejail software versions starting with 0.9.30.
> Long Term Support (LTS) Firejail branch is not affected by this bug.
> 
> Workaround: Disable overlayfs feature at runtime. In a text editor open /etc/firejail/firejail.config file,
> and set "overlayfs" entry to "no".
> 
>       $ grep overlayfs /etc/firejail/firejail.config
>       # Enable or disable overlayfs features, default enabled.
>       overlayfs no
> 
> Fix: The bug is fixed in Firejail version 0.9.64.4
> 
> GitHub commit: (file configure.ac)
> https://github.com/netblue30/firejail/commit/97d8a03cad19501f017587cc4e47d8418273834b
> 
> Credit:  Security researcher Roman Fiedler analyzed the code and discovered the vulnerability.
> Functional PoC exploit code was provided to Firejail development team.
> A description of the problem is here on Roman's blog:
> 
> https://unparalleled.eu/publications/2021/advisory-unpar-2021-0.txt
> https://unparalleled.eu/blog/2021/20210208-rigged-race-against-firejail-for-local-root/

**Notify maintainers**

@7c6f434c 

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.92-hardened1, NixOS, 20.09.2718.e4adbfbab8a (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos-20.09.2996.002c001ab6f"`
 - channels(user): `"unstable-21.03pre267299.5ff6700bb82"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - firejail
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by stigtsp**
###### Motivation for this change

Backport of #112439

Fixed CVE-2021-26910 #112423

Cc: @NixOS/nixos-release-managers @mweinelt 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by mweinelt — APPROVED** [review]
LGTM. Tests still work. Closes #112423.

**Comment by stigtsp** [comment]
This backport misses ffeb2af9209b25c367fd39a7f693188e3715e51b, will cherry-pick this in.

**Comment by stigtsp** [comment]
@GrahamcOfBorg build firejail.tests

**Comment by cole-h** [comment]
@ofborg eval
@ofborg test firejail

File: pkgs/os-specific/linux/firejail/default.nix [hunk 1]
```diff
@@ -1,36 +1,19 @@
...
+  version = "0.9.64.4";
+
+  src = fetchFromGitHub {
+    owner = "netblue30";
+    repo = "firejail";
+    rev = version;
+    sha256 = "sha256-q/XL8cznHlUXdubUEptEAVma1jRUqFb5XcLAV0RVCzs=";
   };
 
+  buildInputs = [ which ];
```
Comment (jonringer) [inline pkgs/os-specific/linux/firejail/default.nix:14]: not really suitable for a backport, but more correct. Unless it's linking against which...
```suggestion
  nativeBuildInputs = [ which ];
```

**Review by jonringer — APPROVED** [review]
diff LGTM

**Pull request merged**